### PR TITLE
Hudi connector flink job implementation

### PR DESCRIPTION
**Pull request by sowmya-dixit**

### CODE REVIEW

File: pipeline/hudi-connector/src/main/scala/org/sunbird/obsrv/streaming/HudiConnectorStreamTask.scala [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+    if (config.hmsEnabled) {
+      conf.setBoolean("hive_sync.enabled", config.hmsEnabled)
+      conf.setString(FlinkOptions.HIVE_SYNC_DB.key(), config.hmsDatabaseName)
+      conf.setString("hive_sync.username", config.hmsUsername)
+      conf.setString("hive_sync.password", config.hmsPassword)
+      conf.setString("hive_sync.mode", "hms")
+      conf.setBoolean("hive_sync.use_jdbc", false)
+      conf.setString(FlinkOptions.HIVE_SYNC_METASTORE_URIS.key(), config.hmsURI)
+      conf.setString("hoodie.fs.atomic_creation.support", "s3a")
```
Comment (ravismula) [inline pipeline/hudi-connector/src/main/scala/org/sunbird/obsrv/streaming/HudiConnectorStreamTask.scala:126]: lines: 111, 126: need to load from config?

File: pipeline/hudi-connector/src/main/scala/org/sunbird/obsrv/streaming/HudiConnectorConfig.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  val hudiBasePath: String = config.getString("hudi.table.base.path")
+  val hudiCompactionEnabled: Boolean = config.getBoolean("hudi.compaction.enabled")
+  val hudiWriteTasks: Int = config.getInt("hudi.write.tasks")
+
+  val hmsEnabled: Boolean = if (config.hasPath("hudi.hms.enabled")) config.getBoolean("hudi.hms.enabled") else false
+  val hmsUsername: String = config.getString("hudi.hms.database.username")
+  val hmsPassword: String = config.getString("hudi.hms.database.password")
+  val hmsDatabaseName: String = config.getString("hudi.hms.database.name")
+  val hmsURI: String = config.getString("hudi.hms.uri")
```
Comment (ravismula) [inline pipeline/hudi-connector/src/main/scala/org/sunbird/obsrv/streaming/HudiConnectorConfig.scala:51]: are we expecting these values, even if hms is disabled?

**Review by ravismula — APPROVED** [review]
need to add hudi-connector-image to GitHub actions as well

File: pipeline/hudi-connector/src/main/scala/org/sunbird/obsrv/streaming/HudiConnectorConfig.scala (hunk 2 above)
Comment (anandp504) [reply pipeline/hudi-connector/src/main/scala/org/sunbird/obsrv/streaming/HudiConnectorConfig.scala:51]: @ravismula  These should be optional. However, we are never going to deploy the flink job without HMS.

File: pipeline/hudi-connector/src/main/scala/org/sunbird/obsrv/streaming/HudiConnectorStreamTask.scala (hunk 1 above)
Comment (anandp504) [reply pipeline/hudi-connector/src/main/scala/org/sunbird/obsrv/streaming/HudiConnectorStreamTask.scala:126]: @ravismula These are more or less mandatory configurations. The atomic creation support may be is applicable only on S3.

**Review by anandp504 — APPROVED** [review]
Approved 👍

**Pull request merged**